hummingbot/client/command/gateway_command.py

```diff
@@ -385,6 +390,69 @@ async def _prompt_for_wallet_address(
         return wallet_address, additional_prompt_values
 
+    async def _gateway_balance(self):
+        # get gateway connections
+        gateway_connections_conf: List[Dict[str, str]] = GatewayConnectionSetting.load()
+        if len(gateway_connections_conf) < 1:
+            self.notify("No existing gateway connection.\n")
+            return
+
+        self.notify("Updating gateway balances and allowances, please wait...")
+
+        chain_network_address_to_connector_tokens: Dict(Tuple[str, str, str], Dict[str, List[str]]) = {}
+        for conf in gateway_connections_conf:
+            tokens_str = conf.get('tokens', '')
+            if tokens_str == "":
+                tokens = []
+            else:
+                tokens = [token.strip() for token in tokens_str.split(',')]
+            if (conf["chain"], conf["network"], conf["wallet_address"]) not in chain_network_address_to_connector_tokens:
+                chain_network_address_to_connector_tokens[(conf["chain"], conf["network"], conf["wallet_address"])] = {conf["connector"]: tokens}
+            else:
+                chain_network_address_to_connector_tokens[(conf["chain"], conf["network"], conf["wallet_address"])][conf["connector"]] = tokens
+
+        # get balances and allowances for each chain-network-address tuple and display them in a table format
+        for chain_network_address, connector_to_tokens in chain_network_address_to_connector_tokens.items():
+
+            self.notify(f"\nwallet: {chain_network_address[2]}")
+            self.notify(f"chain-network: {chain_network_address[0]}-{chain_network_address[1]}")
+
+            connectors: List[str] = list(connector_to_tokens.keys())
+
+            all_tokens = list(set(list(itertools.chain.from_iterable(connector_to_tokens.values()))))
+            native_token: str = native_tokens[chain_network_address[0]]
+            all_tokens = list(set(all_tokens + [native_token]))
+
+            token_balances: Dict[str, Any] = await self._get_gateway_instance().get_balances(
+                *chain_network_address, all_tokens
+            )
+            token_balances = token_balances.get("balances", {})
+            balances: List[str] = [str(round(Decimal(token_balances.get(token, "0")), 4)) for token in all_tokens]
+
+            connector_to_token_allowances: Dict[str, Dict[str, Any]] = {}
+            for connector, tokens in connector_to_tokens.items():
+                token_allowances: Dict[str, Any] = await self._get_gateway_instance().get_allowances(
+                    *chain_network_address, tokens, connector
+                )
+                connector_to_token_allowances[connector] = token_allowances.get("approvals", {})
+
+            raw_data: List[List[str]] = [all_tokens, balances]
+            for connector in connectors:
+                raw_data.append([str(round(Decimal(connector_to_token_allowances[connector].get(token, "NaN")), 4)) for token in all_tokens])
```

Comment:
This line is throwing an error when I run `gateway balance`:
```
2023-07-25 17:18:12,850 - 49239 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: [<class 'decimal.InvalidOperation'>]
Traceback (most recent call last):
  File "/Users/feng1/Code/hummingbot/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "/Users/feng1/Code/hummingbot/hummingbot/client/command/gateway_command.py", line 441, in _gateway_balance
    raw_data.append([str(round(Decimal(connector_to_token_allowances[connector].get(token, "NaN")), 4)) for token in all_tokens])
  File "/Users/feng1/Code/hummingbot/hummingbot/client/command/gateway_command.py", line 441, in <listcomp>
    raw_data.append([str(round(Decimal(connector_to_token_allowances[connector].get(token, "NaN")), 4)) for token in all_tokens])
decimal.InvalidOperation: [<class 'decimal.InvalidOperation'>]
```

Generally, writing code where many operations are in one line is not good practice, since it's complex to debug and difficult to maintain. Please break out this line into multiple lines, each that does a single operation.

Reply:
there's likely an issue with certain tokens or chains - if you separate this line, it'll be easier to find root cause